rfd/0026/README.md

```diff
@@ -1274,38 +1288,32 @@ an appropriate error message when used on shared volumes zones.
 Machines acting as shared volumes' storage zones will have the value `nfsserver`
 for their `smartdc_role` property. To know of this new `smartdc_role` value is used by CloudAPI to prevent users from performing operations on these storage VMs, refer to
 
-#### New internal `required_nfs_volumes` property on VM objects
+#### New `volumes` property on VM objects
 
-A VM object that represents a Docker container mounting a shared volumes will
-store a reference to that volume in a new _indexable_ property named
-`required_nfs_volumes`.
+A VM object that represents a container mounting a shared volumes will store a
+reference to that volume in a new _indexable_ property named `volumes`.
 
-This will allow VOLAPI to perform an indexed search on VMAPI to determine what
-uses a given volume when handling operations that depends on a volume's usage,
-e.g a `DeleteVolume` request.
+This property will also be exposed both at VMAPI and CloudAPI where it can be
```

Comment:
Currently, at the VMAPI level when creating a VM, we pass the information related to NFS volumes that need to be mounted when the VM starts in the `docker:nfsvolumes` metadata.

Is this suggesting that we'd change that to pass that information as a "regular" `volumes` parameter? It seems like a good idea to me. We could then only create the `triton:volumes` metadata (and the same data in `docker:nfsvolumes` to not break docker automounting volumes in older platforms) in the provisioning workflow.

However, if we add a new `volume` parameter for VMAPI's `CreateVm` endpoint, we'd lose the (currently implicit) information about the type of volume to be created (which currently is only `tritonnfs`). So we'd need to add that.

The other thing we'd need to add would be the "mounting mode" (read-only, read-write, etc.) that `docker volume run -v` supports, and that we'd probably want to support with CloudAPI too.